src/orquestra/sdk/_base/_driver/_client.py

```diff
@@ -119,33 +132,33 @@ class DriverClient:
     Client for interacting with the Workflow Driver API via HTTP.
     """
 
-    def __init__(self, base_uri: str, session: requests.Session):
-        self._base_uri = base_uri
+    def __init__(self, session: requests.Session, uri_provider: ExternalUriProvider):
+        self._uri_provider = uri_provider
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** For ‏Either now or next PR:
We want to be able to pass any `URIProvider`, so this should be a protocol or something.

Reply:
We just though of either protocol in this file and implementations is separate, or just Union type. Either we thought to leave it up for the actual implementation 👍